### PR TITLE
NOISSUE: Limit user to have single relation with things, groups, channels and domains

### PR DESCRIPTION
**Pull request by arvindh123**
Signed-off-by: Arvindh <arvindh91@gmail.com>

Pull request title should be `MG-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](https://github.com/absmach/magistrala/blob/master/CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
- Limit user to have single relation with things, groups, channels and domains

### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: auth/spicedb/policies.go [hunk 1]
```diff
@@ -422,6 +422,19 @@ func (pa *policyAgent) addPolicyPreCondition(ctx context.Context, pr auth.Policy
 
 func (pa *policyAgent) userGroupPreConditions(ctx context.Context, pr auth.PolicyReq) ([]*v1.Precondition, error) {
 	var preconds []*v1.Precondition
+
+	// user should not any relation with group
```
Comment (rodneyosodo) [inline auth/spicedb/policies.go:426]: grammar. Should user have or have not?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: auth/spicedb/policies.go [hunk 2]
```diff
@@ -482,6 +495,20 @@ func (pa *policyAgent) userGroupPreConditions(ctx context.Context, pr auth.Polic
 
 func (pa *policyAgent) userThingPreConditions(ctx context.Context, pr auth.PolicyReq) ([]*v1.Precondition, error) {
 	var preconds []*v1.Precondition
+
+	// user should not any relation with thing
```
Comment (ianmuchyri) [inline auth/spicedb/policies.go:499]: same here, `user should not have any relation with thing`

File: auth/spicedb/policies.go (hunk 1 above)
Comment (arvindh123) [reply auth/spicedb/policies.go]: done

File: auth/spicedb/policies.go (hunk 2 above)
Comment (arvindh123) [reply auth/spicedb/policies.go]: done

**Review by ianmuchyri — APPROVED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**